src/stages/appflow-ingestion.ts

```diff
@@ -0,0 +1,136 @@
+import * as path from "path";
+import { Duration } from "aws-cdk-lib";
+import * as appflow from "aws-cdk-lib/aws-appflow";
+import * as events from "aws-cdk-lib/aws-events";
+import * as iam from "aws-cdk-lib/aws-iam";
+import * as lambda from "aws-cdk-lib/aws-lambda";
+import * as sfn from "aws-cdk-lib/aws-stepfunctions";
+import * as tasks from "aws-cdk-lib/aws-stepfunctions-tasks";
+import { Construct } from "constructs";
+import { StateMachineStage, StateMachineStageProps } from "../pipelines/stage";
+
+export interface AppFlowIngestionStageProps extends StateMachineStageProps {
+  readonly flowName?: string;
+  readonly flowExecutionStatusCheckPeriod?: Duration;
+  readonly destinationFlowConfig?: appflow.CfnFlow.DestinationFlowConfigProperty;
+  readonly sourceFlowConfig?: appflow.CfnFlow.SourceFlowConfigProperty;
+  readonly flowTasks?: appflow.CfnFlow.TaskProperty[];
+}
+
+export class AppFlowIngestionStage extends StateMachineStage {
+  readonly targets?: events.IRuleTarget[];
+  readonly eventPattern?: events.EventPattern;
+  readonly stateMachine: sfn.StateMachine;
+  readonly flowObject: sfn.CustomState;
+
+  constructor(scope: Construct, id: string, props: AppFlowIngestionStageProps) {
+    super(scope, id, props);
+
+    const { flowName, flowExecutionStatusCheckPeriod, destinationFlowConfig, sourceFlowConfig, flowTasks } = props;
+
+    if (!flowName) {
+      // Check required props for CfnFlow create and except if not provided
+      if (!destinationFlowConfig || !sourceFlowConfig || !flowTasks) {
+        throw new Error(
+          "if 'flowName' is not specified, 'destinationFlowConfig', 'sourceFlowConfig' & 'tasks' are required properties",
+        );
+      }
+      const flow = new appflow.CfnFlow(this, `${id}-flow`, {
+        destinationFlowConfigList: [destinationFlowConfig],
+        flowName: `${id}-flow`,
```

Comment:
I think we should remove the explicit `flowName`. That way CFN will just generate the name.